std/compress/lzss/compress.go

```diff
@@ -210,6 +210,14 @@ func (compressor *Compressor) Compress(d []byte) (c []byte, err error) {
 		return nil, err
 	}
 
+	if compressor.buf.Len() >= len(d)+2 {
+		// compression was not worth it
+		out := compressor.buf.Bytes()
+		out[1] = byte(NoCompression)
```

Comment:
use a clean `writeHeader` or `writeMetadata` func so the logic to read / write that is not dispersed